### PR TITLE
Render help pages

### PR DESCRIPTION
**Pull request by mcgoooo**
I think maybe these two content types could be merged into one
to avoid duplicate code in the front end. With the css i have just followed
the current pattern of repetition, until we tackle it more wholesale.

i changed the gsub's for squish in the govspeak test helper, as it seems more
robust, and a content item was causing leading space problems

### CODE REVIEW

File: app/presenters/govspeak_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class GovspeakPresenter < ContentItemPresenter
+  include Linkable
```
Comment (fofr) [inline app/presenters/govspeak_presenter.rb]: What's `Linkable` used for?

File: app/views/content_items/answer.html.erb [hunk 2]
```diff
@@ -1,15 +1 @@
...
-        title: @content_item.title %>
-    <%= render 'govuk_component/govspeak',
-        content: @content_item.body,
-        direction: page_text_direction,
-        rich_govspeak: true
-    %>
-  </div>
-  <%= render 'shared/related_items', content_item: @content_item  %>
-</div>
+<%= render 'content_items/short_text/just_govspeak' %>
```
Comment (fofr) [inline app/views/content_items/answer.html.erb]: Could we avoid the `short_text` nesting? Technically everything in government-frontend is short text except HTML pubs.

Comment (fofr) [inline app/views/content_items/answer.html.erb]: I think the label is confusing.

File: app/views/content_items/short_text/_just_govspeak.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<% content_for :simple_header, true %>
```
Comment (fofr) [inline app/views/content_items/short_text/_just_govspeak.html.erb]: Technically this isn't _just govspeak_, it's a title, body and related links.
Maybe body_with_related_links, or something similar? This would be consistent with the names of other partials.

File: app/presenters/help_page_presenter.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+class HelpPagePresenter < GovspeakPresenter
```
Comment (fofr) [inline app/presenters/help_page_presenter.rb]: I think GovspeakPresenter is misleading. Many formats present govspeak.

Perhaps a `body` module that has one method:
```ruby
def body
  content_item["details"]["body"]
end
```

Then the two presenters include that module. Presenters currently use modules rather than inheritance.

Comment (fofr) [inline app/presenters/help_page_presenter.rb]: (I think inheritance could go a long way to cleaning up some of presenters, but I'd like to consider more formats when investigation that)

File: test/test_helper.rb [hunk 5]
```diff
@@ -72,7 +72,7 @@ def assert_has_component_title(title)
 
   def assert_has_component_govspeak(content)
     within_component_govspeak do
-      assert_equal content.gsub(/\s+/, ' '), JSON.parse(page.text).fetch("content").gsub(/\s+/, ' ')
+      assert_equal content.squish, JSON.parse(page.text).fetch("content").squish
```
Comment (fofr) [inline test/test_helper.rb]: :+1:

File: app/presenters/help_page_presenter.rb (hunk 4 above)
Comment (mcgoooo) [inline app/presenters/help_page_presenter.rb]: The naming, i agree with, but overall, if we want to clean this up, perhaps we should be putting the tickets in now?

File: app/presenters/govspeak_presenter.rb (hunk 1 above)
Comment (mcgoooo) [inline app/presenters/govspeak_presenter.rb]: oversite, i think, will double check

File: app/views/content_items/answer.html.erb (hunk 2 above)
Comment (mcgoooo) [inline app/views/content_items/answer.html.erb]: 👍

File: app/presenters/help_page_presenter.rb (hunk 4 above)
Comment (mcgoooo) [inline app/presenters/help_page_presenter.rb]: overall, the way they have done modularity in here has not worked out well imo

Comment (fofr) [inline app/presenters/help_page_presenter.rb]: >  perhaps we should be putting the tickets in now?
Never a reason not to put a ticket in the backlog for discussion and prioritising.

File: app/controllers/content_items_controller.rb [hunk 6]
```diff
@@ -26,8 +26,6 @@ def present(content_item)
     presenter_name = content_item['schema_name'].classify + 'Presenter'
     presenter_class = Object.const_get(presenter_name)
     presenter_class.new(content_item)
-  rescue NameError
```
Comment (fofr) [inline app/controllers/content_items_controller.rb]: Why have you remove this?

File: app/presenters/updated.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+module Updated
```
Comment (fofr) [inline app/presenters/updated.rb]: Could we call this `LastUpdated`?

File: test/integration/answer_test.rb [hunk 8]
```diff
@@ -5,6 +5,7 @@ class AnswerTest < ActionDispatch::IntegrationTest
     setup_and_visit_content_item('answer')
     assert page.has_text?(@content_item["title"])
     assert_has_component_govspeak(@content_item["details"]["body"])
+    assert page.has_text?("Last Updated: 28 Mai 2015")
```
Comment (fofr) [inline test/integration/answer_test.rb]: Why is this `Mai` and not `May`?

File: app/controllers/content_items_controller.rb (hunk 6 above)
Comment (mcgoooo) [inline app/controllers/content_items_controller.rb]: because it does nothing and masks real errors, everytime you have an error anywhere in your presenter, it gives the reraised error, which is not helpful

File: test/integration/answer_test.rb (hunk 8 above)
Comment (mcgoooo) [inline test/integration/answer_test.rb]: it's in welsh

File: app/controllers/content_items_controller.rb (hunk 6 above)
Comment (fofr) [inline app/controllers/content_items_controller.rb]: This sort of change should be in a separate commit with a clear reason.
I don't think it belongs in this PR.

File: app/views/content_items/_body_with_related_links.html.erb [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  <div class="column-two-thirds">
+    <%= render 'govuk_component/title',
+        title: @content_item.title %>
+    <%= render 'govuk_component/govspeak',
+        content: @content_item.body,
+        direction: page_text_direction %>
+    <div class="meta-data group">
+      <% if @content_item.last_updated %>
+        <p class="modified-date">
+          <%= t 'common.last_updated' %>: <%= @content_item.last_updated %>
```
Comment (fofr) [inline app/views/content_items/_body_with_related_links.html.erb]: The strings for `'common.last_updated'` haven't been added to the locale files.

File: app/controllers/content_items_controller.rb (hunk 6 above)
Comment (mcgoooo) [inline app/controllers/content_items_controller.rb]: i can put it in a seperate commit, but you think it needs a whole seperate PR, and deploy for something like that, that only effects development?

File: test/integration/answer_test.rb (hunk 8 above)
Comment (fofr) [inline test/integration/answer_test.rb]: Should `Last Updated:` be welsh too?
https://github.com/alphagov/frontend/blob/master/config/locales/cy.yml#L3

File: test/presenters/help_page_presenter_test.rb [hunk 10]
```diff
@@ -0,0 +1,17 @@
+require 'presenter_test_helper'
+
+class HelpPagePresenterTest < PresenterTestCase
+  include ActionView::Helpers::UrlHelper
```
Comment (fofr) [inline test/presenters/help_page_presenter_test.rb]: Whats the `UrlHelper` needed for?

File: app/views/content_items/_body_with_related_links.html.erb (hunk 9 above)
Comment (mcgoooo) [inline app/views/content_items/_body_with_related_links.html.erb]: i thought the locales came in from some automagic through the static/slimmer nastiness?

Comment (fofr) [inline app/views/content_items/_body_with_related_links.html.erb]: I think that will switch the text from "Last Updated" to "Last updated" (lowercase U):
https://github.com/alphagov/frontend/blob/master/config/locales/en.yml#L3

File: app/controllers/content_items_controller.rb (hunk 6 above)
Comment (fofr) [inline app/controllers/content_items_controller.rb]: Looking at Errbit now, I can see this error is being raised on production:
eg `No support for schema "special_route"`

Worth some separate investigation.

File: app/assets/stylesheets/views/_help-page.scss [hunk 11]
```diff
@@ -0,0 +1,9 @@
+.help-page {
+  .add-title-margin {
+    @include responsive-top-margin;
+  }
+
+  .meta-data {
```
Comment (fofr) [inline app/assets/stylesheets/views/_help-page.scss]: Could this be `last-updated` too? `meta-data` risks being confused with the metadata component.

File: app/assets/stylesheets/views/_help-page.scss [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+    padding-bottom: 15px;
```
Comment (fofr) [inline app/assets/stylesheets/views/_help-page.scss]: `15px` should be `$gutter-half`

File: app/views/content_items/_body_with_related_links.html.erb [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+<%= content_for :title, @content_item.title %>
+
+<div class="grid-row sidebar-with-body">
+  <div class="column-two-thirds">
+    <%= render 'govuk_component/title',
+        title: @content_item.title %>
+    <%= render 'govuk_component/govspeak',
+        content: @content_item.body,
+        direction: page_text_direction %>
+    <div class="meta-data group">
```
Comment (fofr) [inline app/views/content_items/_body_with_related_links.html.erb]: Why do we need the `group` class?

The `<p class="modified-date">` container should be enough. We can remove the `div` container completely.

**Comment by fofr** [comment]
Can you include some screenshots of your latest changes?
From what I can tell there are no styles to render `Last updated` in a lighter colour.

File: app/views/content_items/_body_with_related_links.html.erb (hunk 13 above)
Comment (mcgoooo) [inline app/views/content_items/_body_with_related_links.html.erb]: tbh, this was copied directly from frontend, that is what is there

**Comment by fofr** [comment]
The title is missing the 'Help' context:
screen shot 2017-04-26 at 17 19 16

**Comment by mcgoooo** [comment]
@fofr already discussed with mia, not really needed. the context

**Comment by fofr** [comment]
The value of the last updated date appears to be different, despite both pointing at the same content item (this is from https://www.gov.uk/help/cookies):
screen shot 2017-04-26 at 17 20 20

**Comment by mcgoooo** [comment]
@fofr i also noticed that, and i believe the public_updated_at would be the correct one to use, if not, the schema is incorrect

**Comment by fofr** [comment]
@mcgoooo It depends on how Publisher has been wired up, I'd ask some of the folks who migrated it to confirm or fix behaviour.

**Comment by mcgoooo** [comment]
@fofr can you tag them in this pull please, i have no idea who they are

**Comment by mcgoooo** [comment]
we are going to await the public updated at to be fixed, for the moment we shall remove it
